File: public/js/service/info.js

import { get_info } from "../api/info.js";

const store = Vue.reactive({});

export const fetch_info = () => get_info().then(i => Object.assign(store, i));

export const get_baseurl = () => store.baseurl;
export const get_stage = () => store.stage;

export const get_github_login = () => store.github_login;
export const get_discord_login = () => store.discord_login;
export const get_mesehub_login = () => store.mesehub_login;
export const get_cdb_login = () => store.cdb_login;

export const get_hostingdomain_suffix = () => store.hostingdomain_suffix;
export const get_max_balance = () => store.max_balance;
export const get_zahlsch_enabled = () => store.zahlsch_enabled;
export const get_coinbase_enabled = () => store.coinbase_enabled;

File: types/info.go
package types

type Info struct {
	BaseURL             string `json:"baseurl"`
	HostingDomainSuffix string `json:"hostingdomain_suffix"`
	Stage               string `json:"stage"`
	CoinbaseEnabled     bool   `json:"coinbase_enabled"`
	WalleeEnabled       bool   `json:"wallee_enabled"`
	ZahlschEnabled      bool   `json:"zahlsch_enabled"`
	ZahlschPageID       string `json:"zahlsch_pageid"`
	ZahlschUser         string `json:"zahlsch_user"`
	GithubLogin         string `json:"github_login"`
	DiscordLogin        string `json:"discord_login"`
	MesehubLogin        string `json:"mesehub_login"`
	CDBLogin            string `json:"cdb_login"`
	MaxBalance          int    `json:"max_balance"` //max balance in eurocents
}

func NewInfo(cfg *Config) *Info {
	return &Info{
		BaseURL:             cfg.BaseURL,
		HostingDomainSuffix: cfg.HostingDomainSuffix,
		Stage:               cfg.Stage,
		CoinbaseEnabled:     cfg.CoinbaseEnabled,
		WalleeEnabled:       cfg.WalleeEnabled,
		ZahlschEnabled:      cfg.ZahlschEnabled,
		ZahlschPageID:       cfg.ZahlschPageID,
		ZahlschUser:         cfg.ZahlschUser,
		GithubLogin:         cfg.GithubOauthConfig.LoginURL,
		DiscordLogin:        cfg.DiscordOauthConfig.LoginURL,
		MesehubLogin:        cfg.MesehubOauthConfig.LoginURL,
		CDBLogin:            cfg.CDBOauthConfig.LoginURL,
		MaxBalance:          cfg.MaxBalance,
	}
}
